sim/druid/item_sets_pve_phase_4.go

```diff
@@ -240,9 +237,54 @@ var ItemSetCenarionRage = core.NewItemSet(core.ItemSet{
 	},
 })
 
+// Reduces the cooldown of Enrage by 30 sec and it no longer reduces your armor.
+func (druid *Druid) applyT1Guardian4PBonus() {
+	label := "S03 - Item - T1 - Druid - Guardian 4P Bonus"
+	if druid.HasAura(label) {
+		return
+	}
+
+	core.MakePermanent(druid.RegisterAura(core.Aura{
+		ActionID: core.ActionID{SpellID: 456328}, // Tracking in APL
+		Label:    label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			druid.Enrage.CD.FlatModifier -= time.Second * 30
+
+			oldOnGain := druid.EnrageAura.OnGain
+			druid.EnrageAura.OnGain = func(aura *core.Aura, sim *core.Simulation) {
+				preArmor := druid.GetStat(stats.Armor)
+				oldOnGain(aura, sim)
+				postArmor := druid.GetStat(stats.Armor)
+				druid.AddStatDynamic(sim, stats.Armor, preArmor-postArmor)
```

Comment:
Rather than adding stats dynamically here, you could make the enrage armor reduction an attribute on the Druid struct, then set it to 0 in this `OnInit`. That should prevent you from having to override the `OnGain` like this